compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -905,7 +905,9 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             trace!(
                 "deallocating local {:?}: {:?}",
                 local,
-                self.dump_alloc(ptr.provenance.unwrap().get_alloc_id())
+                // We should only be able to deallocate pointers that can
+                // return their `alloc_id`.
```

Comment:
```suggestion
                // Locals always have a `alloc_id` (they are never the result of a int2ptr).
```
It is somewhat annoying that we cannot represent this in the type though... that's why we need so many more `Option` now. :/